packages/fxa-auth-server/lib/oauth/keys.ts

```diff
@@ -139,18 +138,18 @@ exports.generatePrivateKey = function generatePrivateKey() {
   const now = new Date();
   const pubKeyFingerprint = crypto
     .createHash('sha256')
-    .update(kp.publicKey)
+    .update(kp.publicKey.toString())
     .digest('hex')
     .slice(0, 8);
-  const privKey = Object.assign(pem2jwk(kp.privateKey), {
+  const privKey = Object.assign(pem2jwk(kp.privateKey.toString()), {
     kid:
       now.toISOString().slice(0, 10).replace(/-/g, '') +
       '-' +
       pubKeyFingerprint,
     alg: 'RS256',
     use: 'sig',
     // Timestamp to nearest hour; consumers don't need to know the precise time.
-    'fxa-createdAt': Math.floor(now / 1000 / 3600) * 3600,
+    'fxa-createdAt': Math.floor(now.getTime() / 1000 / 3600) * 3600,
```

Comment:
interesting that we need to do this.

Reply:
TypeScript refuses to compile it since you can't divide the Date object, but JS of course implicitly converts it to the number like `getTime()` does.

Reply:
This will be hard to <kbd>grep</kbd> for&hellip; but big +1 to slowly converting everything to TypeScript.
I think this script will have the same issue, eventually:

https://github.com/mozilla/fxa/blob/60fa969bc2fc654d7238cab7a990a3f7a977493f/packages/fxa-auth-server/scripts/gen_keys.js#L53-L65

```sh
git grep -n "now /" | cat

packages/fxa-auth-server/lib/log.js:211:  data.ts = data.ts || now / 1000; // Convert to float seconds
packages/fxa-auth-server/scripts/gen_keys.js:63:  key['fxa-createdAt'] = Math.round(now / 1000 / 3600) * 3600;
packages/fxa-content-server/app/scripts/lib/crypto/totp.js:19:  cv.setUint32(4, Math.floor(now / 30000), false);
packages/fxa-event-broker/src/queueworker/queueworker.service.spec.ts:25:  ts: now / 1000,
packages/fxa-event-broker/src/queueworker/queueworker.service.spec.ts:88:  ts: now / 1000,
packages/fxa-event-broker/src/queueworker/queueworker.service.spec.ts:295:          ts: now / 1000,
```

FILE | CODE | NOTES
:----|:---|:---
packages/fxa-auth-server/lib/log.js:211: | `const now = Date.now();` | `now` is numeric.
packages/fxa-auth-server/scripts/gen_keys.js:63: | `const now = new Date();` | `now` is a Date object.
packages/fxa-content-server/app/scripts/lib/crypto/totp.js:19: | `now = Date.now()` | `now` is an argument (but default _should_ be numeric).
packages/fxa-event-broker/src/queueworker/queueworker.service.spec.ts:25:<br><br>packages/fxa-event-broker/src/queueworker/queueworker.service.spec.ts:88:<br><br>packages/fxa-event-broker/src/queueworker/queueworker.service.spec.ts:295: | `const now = Date.now();` | `now` is numeric.

Reply:
Filed as https://mozilla-hub.atlassian.net/browse/FXA-9646 with a few bonus greps.